### ISSUE TITLE
Questions about SynthMorph validation phase

### ISSUE THREAD

**Issue by nohsoocheol** [voxelmorph/voxelmorph#420]
**Task** (what are you trying to do/register?)

SynthMorph validation phase

**What have you tried**

SynthMorph validation phase

**Details of experiments**

First of all, thank you for sharing codes and papers.
I have reimplemented SynthMorph codes based on pytorch and I'm wondering how to use a dataset in the validation phase.
In SynthMorph paper, using 10 registration pairs is mentioned, but I can't find any information about images.
Did you just use real images or synthesized images from labels? If you used synthesized images, then did you fix seed numbers for the label_to_image generator to consistently generate same images among epochs?

Thanks again for sharing your great work!

**Comment by adalca** [comment]
@mu40 can comment best here, but you can probably use data like https://github.com/adalca/medical-datasets/blob/master/neurite-oasis.md

**Comment by mu40** [comment]
For validation, I suppose you will want to use real acquired MR images, assuming this is what you plan to register down the line. In the SynthMorph paper, we test several public datasets and measure accuracy by comparing label maps after applying the transforms estimated from the corresponding images.

While you can get label maps by segmenting images using a method of choice (e.g. SynthSeg, SAMSEG, FreeSurfer), it may be more convenient to use a dataset that already includes segmentations, just as the one @adalca suggested.

**Comment by nohsoocheol** [comment]
Thank to both of you for the answers and the link!

**Comment by aungriah** [comment]
@nohsoocheol Did you manage to get a similar performance with PyTorch as with Tensorflow when training SynthMorph? I have been trying to replicated the demo in PyTorch but I haven't been able to get the performance to be as good, even after translating individual Tensorflow Layers...
